core/src/test/java/org/itsallcode/openfasttrace/core/serviceloader/ServiceLoaderFactoryTest.java

```diff
@@ -0,0 +1,130 @@
+package org.itsallcode.openfasttrace.core.serviceloader;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import org.itsallcode.openfasttrace.api.report.ReporterFactory;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.opentest4j.MultipleFailuresError;
+
+// [utest->dsn~plugins.loading~1]
+class ServiceLoaderFactoryTest
+{
+    @TempDir
+    Path tempDir;
+
+    @Test
+    void noPluginJar()
+    {
+        assertThat(testee().createLoader(ReporterFactory.class).load().toList(), empty());
+    }
+
+    private ServiceLoaderFactory testee()
+    {
+        return new ServiceLoaderFactory(tempDir, true);
+    }
+
+    @Test
+    void findServiceSkipCurrentClassLoader() throws IOException
+    {
+        Files.delete(tempDir);
+        assertThat(new ServiceLoaderFactory(tempDir, false).findServiceOrigins(), empty());
+    }
+
+    @Test
+    void findServiceMissingParentDir() throws IOException
+    {
+        Files.delete(tempDir);
+        final List<ServiceOrigin> origins = testee().findServiceOrigins();
+        assertNoPlugins(origins);
+    }
+
+    private void assertNoPlugins(final List<ServiceOrigin> origins) throws MultipleFailuresError
+    {
+        assertAll(() -> assertThat(origins, hasSize(1)),
+                () -> assertThat(origins.get(0).getClassLoader().getName(), equalTo("app")));
+    }
+
+    @Test
+    void findServiceOriginsPluginDir()
```

Comment:
If I look at the code below, this tests behavior on an *empty* plugins parent dir.